### PR TITLE
Add RSA_KEY_SIZE flag to cli test script

### LINKED ISSUE #112: Request to add override option for `parsec-cli-test.sh` key length (bits)
Feature request:

Some platforms are very slow computing RSA 2048 bits keys (RSA or ECC) or simply not capable of generating the default where the test fails, Is it possible to have the option to override in `parsec-cli-tests.sh` to use a different key length bits to enable testing?

### PR DESCRIPTION
**Pull request by gowthamsk-arm**
This flag can be used for testing parsec on slower platforms where RSA 2048 key operations are time consuming.

Closes #112

Signed-off-by: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>

### CODE REVIEW

**Comment by tgonzalezorlandoarm** [comment]
I was wondering if we could start doing something like:

**Comment by tgonzalezorlandoarm** [comment]
> I was wondering if we could start doing something like:
> 
> `tests/parsec-cli-tests.sh: Add RSA_KEY_SIZE flag`

See for example: https://github.com/parallaxsecond/parsec-tool/commit/2c78cc39707f52597639c4479a683441f75e0a1b

**Comment by tgonzalezorlandoarm** [comment]
> I was wondering if we could start doing something like:
> 
> `tests/parsec-cli-tests.sh: Add RSA_KEY_SIZE flag`

To be clearer, I'm talking about the commit message here.

**Comment by gowthamsk-arm** [comment]
Sounds good. Will update the commit.

File: tests/parsec-cli-tests.sh [hunk 1]
```diff
@@ -286,6 +298,7 @@ Usage: $0 [parameter]
     -N:   Test only the provider with N ID
     --no-oaep: Do not test RSA-OAEP(SHA256) encryption/decryption operations
     --no-v1.5: Do not test RSA-PKCS#1-v1.5 encryption/decryption operations
+    --rsa-key-size: Perform all RSA operations with a fixed key length
```
Comment (tgonzalezorlandoarm) [inline tests/parsec-cli-tests.sh:301]: --rsa-key-size: Perform all RSA operations with a fixed key length with the specified key length?

File: tests/ci.sh [hunk 2]
```diff
@@ -32,3 +32,4 @@ fi
 ./target/debug/parsec-tool --help
 
 PARSEC_TOOL="./target/debug/parsec-tool" tests/parsec-cli-tests.sh -d
+PARSEC_TOOL="./target/debug/parsec-tool" tests/parsec-cli-tests.sh --rsa-key-size 1024
```
Comment (tgonzalezorlandoarm) [inline tests/ci.sh:35]: Maybe add the "-d" option here as well?

File: tests/parsec-cli-tests.sh [hunk 3]
```diff
@@ -277,6 +286,9 @@ while [ "$#" -gt 0 ]; do
         --no-v1.5 )
             NO_PKCS1_V15="true"
         ;;
+        --rsa-key-size )
+            shift; RSA_KEY_SIZE=$1
```
Comment (tgonzalezorlandoarm) [inline tests/parsec-cli-tests.sh:290]: Why the `shift` here?

File: tests/parsec-cli-tests.sh [hunk 4]
```diff
@@ -101,7 +107,7 @@ test_crypto_provider() {
 test_encryption() {
 # $1 - algorithm
     KEY="anta-key-rsa-encrypt"
-    TEST_STR="$(date) Parsec public key encryption test"
+    TEST_STR="$(date) Parsec public key encryption"
```
Comment (tgonzalezorlandoarm) [inline tests/parsec-cli-tests.sh:110]: Why this change?

**Review by tgonzalezorlandoarm — CHANGES_REQUESTED** [review]

File: tests/parsec-cli-tests.sh (hunk 1 above)
Comment (gowthamsk-arm) [reply tests/parsec-cli-tests.sh:301]: Sorry I didnt get it. Do you want me to reword it to 
`--rsa-key-size: Perform all RSA operations with the specified key length`?

File: tests/parsec-cli-tests.sh (hunk 3 above)
Comment (gowthamsk-arm) [reply tests/parsec-cli-tests.sh:290]: To extract the value of RSA_KEY_SIZE

File: tests/parsec-cli-tests.sh (hunk 4 above)
Comment (gowthamsk-arm) [reply tests/parsec-cli-tests.sh:110]: For a RSA 1024 key, the string to be encrypted + its padding should be smaller than the key size.

Comment (gowthamsk-arm) [reply tests/parsec-cli-tests.sh:110]: Without this change the script fails.

Comment (tgonzalezorlandoarm) [reply tests/parsec-cli-tests.sh:110]: Oh, my bad. Thanks for the clarification!

File: tests/parsec-cli-tests.sh (hunk 1 above)
Comment (tgonzalezorlandoarm) [reply tests/parsec-cli-tests.sh:301]: yes, if you agree

File: tests/parsec-cli-tests.sh [hunk 5]
```diff
@@ -236,13 +242,15 @@ test_csr() {
...
+       key_size="$RSA_KEY_SIZE"
+       key_param="--bits $RSA_KEY_SIZE"
+    elif [ -n "$1" ]; then
        key_size=$1
        key_param="--bits $1"
     else
-       key_size=${DEFAULT_SIZE}
+       key_size=2048
        key_param=""
     fi
```
[the remark this answers is not in the document]
Comment (gowthamsk-arm) [reply tests/parsec-cli-tests.sh:256]: Added now.

**Review by minosgalanakis — APPROVED** [review]
LGTM
